### PR TITLE
Current user, national, boolean for tsql->hive

### PR DESCRIPTION
**Pull request by 10bas10**
Fixed several issues regarding tranpiling from T-SQL to Spark:
- Removed national identifier in hive and depending dialects as its not supported;
- Fixed type mapping for BIT to BOOLEAN in hive and depending dialects;
- Added support current user

### CODE REVIEW

**Review by GeorgeSittas — APPROVED** [review]
Looks good, left one comment.

File: sqlglot/tokens.py [hunk 1]
```diff
@@ -163,6 +163,7 @@ class TokenType(AutoName):
     CURRENT_ROW = auto()
     CURRENT_TIME = auto()
     CURRENT_TIMESTAMP = auto()
+    CURRENT_USER = auto()
```
[the remark this answers is not in the document]
Comment (GeorgeSittas) [reply sqlglot/tokens.py:166]: Ah, sorry. I missed that you added the token to `NO_PAREN_FUNCTIONS`, which in turn is included in `ID_VAR_TOKENS`.

 Let's add a test where `CURRENT_USER` is used without parentheses too then:

```sql
SELECT CURRENT_USER  -- valid Spark SQL
```

File: tests/dialects/test_spark.py [hunk 2]
```diff
@@ -345,3 +345,13 @@ def test_bool_or(self):
             "SELECT a, LOGICAL_OR(b) FROM table GROUP BY a",
             write={"spark": "SELECT a, BOOL_OR(b) FROM table GROUP BY a"},
         )
+
+    def test_current_user(self):
+        self.validate_all(
+            "CURRENT_USER",
+            write={"spark": "CURRENT_USER()"},
```
Comment (tobymao) [inline tests/dialects/test_spark.py:352]: is this valid? current_user() is a function in spark?

Comment (GeorgeSittas) [reply tests/dialects/test_spark.py:352]: It works with `spark-sql` for me:

```sql
➜  sqlglot git:(main) spark-sql -e "SELECT CURRENT_USER()"
Setting default log level to "WARN".
To adjust logging level use sc.setLogLevel(newLevel). For SparkR, use setLogLevel(newLevel).
23/04/06 18:57:55 WARN NativeCodeLoader: Unable to load native-hadoop library for your platform... using builtin-java classes where applicable
23/04/06 18:57:59 WARN HiveConf: HiveConf of name hive.stats.jdbc.timeout does not exist
23/04/06 18:57:59 WARN HiveConf: HiveConf of name hive.stats.retries.wait does not exist
23/04/06 18:58:02 WARN ObjectStore: Version information not found in metastore. hive.metastore.schema.verification is not enabled so recording the schema version 2.3.0
23/04/06 18:58:02 WARN ObjectStore: setMetaStoreSchemaVersion called but recording version is disabled: version = 2.3.0, comment = Set by MetaStore georgesittas@192.168.0.212
Spark master: local[*], Application Id: local-1680796677723
georgesittas
Time taken: 4.771 seconds, Fetched 1 row(s)
```

Comment (tobymao) [reply tests/dialects/test_spark.py:352]: and what about without paren?

Comment (GeorgeSittas) [reply tests/dialects/test_spark.py:352]: Yep, both are aliases for the same functionality, tried the other one as well earlier.

**Pull request merged**